Assumes Julia is installed

The registry update step assumes julia is installed and fails the job if it's not. 

cc. @giordano

Example: https://github.com/JuliaMath/QuadGK.jl/actions/runs/10640418662/job/29500338448
```
Registries directory exists and is non-empty. Updating any registries
/Users/runner/work/_temp/76e2caee-c42d-4eab-a2b1-ac350daddc95.sh: line 3: julia: command not found
Error: Process completed with exit code 127.
```

--- a/action.yml
+++ b/action.yml
@@ -140,6 +140,7 @@ runs:
     # Pkg operations. So this is only necessary for newer julia versions.
     - name: Update any cached registries
       if: ${{ inputs.cache-registries == 'true' }}
+      continue-on-error: true
       run: |
         if [ -d "${{ steps.paths.outputs.depot }}/registries" ] && [ -n "$(ls -A "${{ steps.paths.outputs.depot }}/registries")" ]; then
           echo "Registries directory exists and is non-empty. Updating any registries"

